File: c#/LeetCode_1528.cs
/*
Runtime: 108 ms, faster than 82.80% of C# online submissions for Shuffle String.
Memory Usage: 27.6 MB, less than 71.56% of C# online submissions for Shuffle String.
 
 */
using System;
using System.Collections.Generic;


namespace UnitTestProject1
{

    public class UnitTest1
    {
        public static void Main()
        {
            var sol = new UnitTest1();

            String s = "codeleet";
            
            //array
            int[] indices = new int[8] { 4, 5, 6, 7, 0, 2, 1, 3 };

            Console.WriteLine(sol.RestoreString(s, indices));
            Console.Read();

        }

        public string RestoreString(string s, int[] indices)
        {
            Dictionary<int, char> dict1 = new Dictionary<int, char>();
            List<Char> result = new List<Char>();
            int index = 0;

            foreach (var i in s) {

                dict1.Add(indices[index], i);
                index += 1;
            }

            Array.Sort(indices);
            foreach (int ii in indices) {
                result.Add(dict1[ii]);
            }
            
            return String.Join("", result);
        }

    }
}

File: JavaScript/LeetCode_1528.js
/*
Runtime: 84 ms, faster than 81.40% of JavaScript online submissions for Shuffle String.
Memory Usage: 39.5 MB, less than 6.64% of JavaScript online submissions for Shuffle String.
*/

let s = "vttqexwqgdc";

let indices = [9,5,8,0,4,3,6,10,1,2,7];

var restoreString = function(s, indices) {
    let index=0;
    let result=[];
    let dict_res={};
    for (let i of s){
        dict_res[parseInt(indices[index])] = i;
        index+=1;
    }

    //數字大小順序 背起來
    for (let ii of indices.sort(function (a, b) {return a - b})){
        result.push(dict_res[ii]);
    }
    return result.join('');
    
};

result = restoreString(s,indices);
console.log(result);











